File: app/src/main/java/com/example/githubrepo/services/event/BusEventMapping.java
package com.example.githubrepo.services.event;

import java.util.HashMap;

import static com.example.githubrepo.services.event.BusEvent.EventType;

/**
 * Created by sarah_neo on 22/02/2017.
 */

public class BusEventMapping {
    // Zone effects

    public static EventType[] RETRIEVE_PRODUCTS = {EventType.REPOS};

    public static HashMap<Class, EventType[]> getZones() {
        HashMap<Class, EventType[]> zones = new HashMap<>();

        //Zone Mapping
        zones.put(LoadReposEvent.class, RETRIEVE_PRODUCTS);

        return zones;
    }
}

File: app/src/main/java/com/example/githubrepo/services/event/BusEventMapping.kt
package com.example.githubrepo.services.event

import java.util.*

object BusEventMapping {
    // Zone effects
    private val RETRIEVE_PRODUCTS = arrayOf(BusEvent.EventType.REPOS)

    //Zone Mapping
    val zones: HashMap<Class<*>, Array<BusEvent.EventType>>
        get() {
            val zones = HashMap<Class<*>, Array<BusEvent.EventType>>()

            //Zone Mapping
            zones[LoadReposEvent::class.java] = RETRIEVE_PRODUCTS
            return zones
        }
}